src/ingestion/sources/redshift.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+from .base import SQLAlchemyExtractor, SourceConfiguration, SQLAlchemySourceDialect, SQLAlchemySource
+from .postgresql import PostgreSQLSourceDialect
+
+class RedshiftSourceConfiguration(SourceConfiguration):
+    @classmethod
+    def validate(cls, configuration):
+        raise NotImplementedError
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "user": { "type": "string" },
+                "password": { "type": "string" },
+                "host": { "type": "string" },
+                "port": { "type": "string" },
+                "database": { "type": "string" },
+                "schema": { "type": "string" },
+            },
+            "secret": [ "password" ],
+        }
+
+    def connection_string(self) -> str:
+        configuration = json.loads(self.configuration)
+
+        return 'redshift+psycopg2://{user}:{password}@{host}:{port}/{database}'.format(
+            user=configuration.get('user'),
+            password=configuration.get('password'),
+            host=configuration.get('host'),
+            port=configuration.get('port'),
+            database=configuration.get('database'),
+            schema=configuration.get('schema'),
+        )
+
+    @property
+    def type(self):
+        return "redshift"
+
+class RedshiftSourceDialect(PostgreSQLSourceDialect):
```

Comment:
Is any override necessary here? or we don't know? if we don't know , let's throw a TODO here.

Reply:
added TODO